v0/destinations/hs/util.js

```diff
@@ -376,6 +379,181 @@ const getEventAndPropertiesFromConfig = (message, destination, payload) => {
+      }
+    });
+  }
+  return { hsSearchId };
+};
+
+const setHsSearchId = (input, id) => {
+  const { message } = input;
+  const resultExternalId = [];
+  if (message.context && message.context.externalId) {
+    externalIdArray = message.context.externalId;
+  }
+  if (externalIdArray) {
+    externalIdArray.forEach(extIdObj => {
+      const { type } = extIdObj;
+      if (type.includes("HS")) {
+        extIdObj.hsSearchId = id;
+      }
+      resultExternalId.push(extIdObj);
+    });
+  }
+  return resultExternalId;
+};
+/**
+ * DOC: https://developers.hubspot.com/docs/api/crm/search
+ * @param {*} inputs
+ * @param {*} destination
+ */
+const getExistingData = async (inputs, destination) => {
+  const { Config } = destination;
+  const values = [];
+  let searchResponse;
+  const updateHubspotIds = [];
+  const firstMessage = inputs[0].message;
+  let objectType = null;
+  let identifierType = null;
+
+  if (firstMessage) {
+    objectType = getDestinationExternalIDInfoForRetl(firstMessage, "HS")
+      .objectType;
+    identifierType = getDestinationExternalIDInfoForRetl(firstMessage, "HS")
+      .identifierType;
+    if (!objectType || !identifierType) {
+      throw new CustomError(
+        "[HS]:: objectType and identifier type not found.",
+        400
+      );
+    }
+  } else {
+    throw new CustomError(
+      "[HS]:: objectType and identifier type not found. ",
+      400
+    );
+  }
+  inputs.map(async input => {
+    const { message } = input;
+    const { destinationExternalId } = getDestinationExternalIDInfoForRetl(
+      message,
+      "HS"
+    );
+    values.push(destinationExternalId);
+  });
+  let requestData = {
+    filterGroups: [
+      {
+        filters: [
+          {
+            propertyName: identifierType,
+            values,
+            operator: "IN"
+          }
+        ]
+      }
+    ],
+    properties: [identifierType],
+    limit: SEARCH_LIMIT_VALUE,
+    after: 0
+  };
+
+  const requestOptions = {
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${Config.accessToken}`
+    }
+  };
+  let checkAfter = 1; // variable to keep checking if we have more results
+  while (checkAfter) {
+    const endpoint = IDENTIFY_CRM_SEARCH_ALL_OBJECTS.replace(
+      ":objectType",
+      objectType
+    );
+
+    const url =
+      Config.authorizationType === "newPrivateAppApi"
+        ? endpoint
+        : `${endpoint}?hapikey=${Config.apiKey}`;
+    searchResponse =
+      Config.authorizationType === "newPrivateAppApi"
+        ? await httpPOST(url, requestData, requestOptions)
+        : await httpPOST(url, requestData);
+    const after = searchResponse.response?.data?.paging?.next?.after | 0;
+
+    requestData.after = after; // assigning to the new value of after
+    checkAfter = after; // assigning to the new value if no after we assign it to 0 and no more calls will take place
+
+    searchResponse = processAxiosResponse(searchResponse);
```

Comment:
We can move this above at line 532 as soon as we get a response and do the processing of `searchResponse` only if we get a 200 status response.